### ISSUE TITLE
When gemset is detected, travis fails to find and run `pod install` due to lack of `bundle exec`

### ISSUE THREAD

**Issue by jcampbell05** [travis-ci/travis-ci#5694]
When travis detects a gemspec, it runs `pod install` without `bundle exec` causing `pod` to not be found, fastlane uses `bundle exec` so `pod install` is found and ran succesfully

**Comment by BanzaiMan** [comment]
We run `pod install` exactly when `Podfile` is found and the condition `! ([[ -f #{pod_dir}/Podfile.lock && -f #{pod_dir}/Pods/Manifest.lock ]] && cmp --silent #{pod_dir}/Podfile.lock #{pod_dir}/Pods/Manifest.lock)` holds (with an appropriate value of `pod_dir`). Your repository seems to require extra considerations that we do not anticipate.

I am not familiar with Fastlane; for now, it is best to override `install` so that you can get your build going.

**Comment by jcampbell05** [comment]
The problem is really simple and quite common.

Travis CI now detects RVM gemsets from .ruby-gemset files. 

If you don't have one it installs the gems in the system location which it searches when you enter commands so things such as pod install work.

When you have a gender it uses bundler to install to vendor/bundler

This directory isn't searched when executing commands. The only way to run gems installed this way is to prefix the gem command with bundler exec.

Therefor if you have a gemset in your repo you can't use the default install command for travis and cocoapods

All tracks needs to do is when it detects a gemset it needs to run bundle exec pod install and not pod install. In fact it may be simpler to always use bundle exec pod install.

When running fastlane I had to do this as originally it wasn't detecting it without the bundle exec. When running fastlane, it is able to run cocoapods for me since it invokes cocoapods via bundle exec

Lots of people use gemsets and they are common so this will effect loads of people and overusing the install command breaks caching.  Pity since this feature is officially supported by travis

Tl;dr

We should run bundle exec pod install instead of just pod install

**Comment by jcampbell05** [comment]
*gemset not gender (can't edit on my phone)

**Comment by jcampbell05** [comment]
*travis not tracks (can't edit on phone)

**Comment by jcampbell05** [comment]
*overridding the install command

**Comment by BanzaiMan** [comment]
You can pull up the Desktop version with the link at the bottom of the page. That lets you edit your comments.

**Comment by jcampbell05** [comment]
I've updated :)

**Comment by jcampbell05** [comment]
@banzaiman I've created a pr to fix this can you have a look

**Comment by jcampbell05** [comment]
@BanzaiMan It's ready to merge.
